File: src/main/java/novemberizing/rx/operator/Just.java
package novemberizing.rx.operator;

import novemberizing.rx.Scheduler;
import novemberizing.rx.task.Task;
import novemberizing.util.Log;

/**
 *
 * @author novemberizing, novemberizing@gmail.com
 * @since 2017. 1. 12.
 */
public class Just<T> extends Single<T> {
    private static final String Tag = "Just";

    @Override
    protected Task<T, T> on(Task<T, T> task) {
        Log.f(Tag, this, task);
        task.out = task.in;
        return task;
    }

    public static void main(String[] args){
        Log.depth(3);
        Log.disable(Log.HEADER | Log.FLOW);
        Operator.Exec(Just(), "1").on(o->Log.i("",o));
        Scheduler.Local().clear();
    }
}

File: src/main/c++/example/rx/observables/just.cc
#include <novemberizing/rx/observable.hh>
#include <novemberizing/rx/subscribers.hh>

#include <string>

using namespace novemberizing::rx;
using namespace novemberizing::util;

int main(void)
{
    Log::o.disable(0xFFFFFFFF);
    Observable<std::string> * observable = Observable<std::string>::Just({"0", "1", "2", "3", "4", "5", "6", "7", "8", "9"});
    observable->subscribe(Subscribers::Just<std::string>("just >"));
    return 0;
}
